debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -598,6 +602,61 @@ public Object getColumnValue(ResultSet rs, int columnIndex, Column column, Table
         }
     }
 
+    // NOTE: fix for DBZ-7359
+    @Override
+    public void setQueryColumnValue(PreparedStatement statement, Column column, int pos, Object value) throws SQLException {
+        boolean isColumnValueSet = false;
+
+        if (column.typeUsesCharset()) {
```

Comment:
Also it might be good to add a `TRACE` message here and another if we did not handle it went via default.

Reply:
I've held off on doing this till we agree on the cleanest implementation of this method.